Tests/SwiftGenKitTests/FilesTests.swift

```diff
@@ -0,0 +1,55 @@
+//
+// SwiftGenKit UnitTests
+// Copyright © 2020 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+import PathKit
+@testable import SwiftGenKit
+import XCTest
+
+final class FilesTests: XCTestCase {
+  func testEmpty() throws {
+    let parser = try Files.Parser()
+
+    let result = parser.stencilContext()
+    XCTDiffContexts(result, expected: "empty", sub: .files)
+  }
+
+  func testDefaults() throws {
+    let parser = try Files.Parser()
+    try parser.searchAndParse(path: Fixtures.directory(sub: .files))
+
+    let result = parser.stencilContext()
+    XCTDiffContexts(result, expected: "defaults", sub: .files)
+  }
+
+  func testMp4s() throws {
+    let parser = try Files.Parser()
+    let path = Fixtures.directory(sub: .files)
```

Comment:
This should use the `searchAndParse` function (like above) with the filter `filterRegex(forExtensions: ["mp4"])`.

Reply:
Hmm. That `searchAndParse` is probably what the CLI uses isn't it? This might be an issue. The reason I used that code instead was to make sure the `relativeTo` parent dir was the actual directory passed in. That's kind of important for creating a file tree relative to a specific directory. The `if/else` pass in different parent directories which throw that off

I'm not sure what the best solution would be. Have an option on the parser? Add some custom handling in the CLI code? Suggestions?

Reply:
Changing this to `searchAndParse`. Even though logically the output from this is not what would be desired, after adding the `relativeTo` option, this would be the designed output